python/amici/petab_objective.py

```diff
@@ -155,6 +159,114 @@ def simulate_petab(
+    :param solver:
+        An AMICI solver. Will use default options if None.
+    :param problem_parameters:
+        Run simulation with these parameters. If None, PEtab `nominalValues`
+        will be used). To be provided as dict,  mapping PEtab problem
+        parameters to SBML IDs.
+    :param simulation_conditions:
+        Result of `petab.get_simulation_conditions`. Can be provided to save
+        time if this has be obtained before.
+        Not required if `edatas` and `parameter_mapping` are provided.
+    :param edatas:
+        Experimental data. Parameters are inserted in-place for simulation.
+    :param parameter_mapping:
+        Optional precomputed PEtab parameter mapping for efficiency, as
+        generated by `create_parameter_mapping`.
+    :param scaled_parameters:
+        If True, problem_parameters are assumed to be on the scale provided
+        in the PEtab parameter table and will be unscaled. If False, they
+        are assumed to be in linear scale.
+    :param log_level:
+        Log level, see :mod:`amici.logging` module.
+    :num_threads:
+        Number of cores to be used for simulations when multi-threading
+    :export_file:
+        Path to (csv) file to which the steady state value should be written
+    :sep:
+        Separator for csv file for export
+    :return:
+        Pandas Dataframe containing
+        * the names of the state variables as row indices,
+        * the indices of the amici edata objects as columns,
+          (**NOTE**: only edatas with timepoints INF will be considered)
+        * the steady state values of the state variables as table content`.
+    """
+
+    # if steady state measurements are available: get the simulation conditions
+    steadystate_conditions = _find_steadystate_conditions(petab_problem)
+
+    if simulation_conditions is not None:
+        # filter simulation conditions for steadystate simulations
+        simulation_conditions = _filter_simulation_conditions(
+            simulation_conditions, steadystate_conditions)
+    else:
+        simulation_conditions = steadystate_conditions
+    # get a list identifying the conditions to be simulated
+    condition_list = _create_condition_list(simulation_conditions)
+
+    if edatas is not None:
+        # edatas overwrites simulation_conditions: Overwrite condition_list.
+        condition_list = [f'edata_{ie}' for ie, edata in enumerate(edatas)
+                          if any(np.isinf(edata.getTimepoints()))]
+        # filter edata objects for steadystate simulations
+        edatas = [edata for edata in edatas
+                  if any(np.isinf(edata.getTimepoints()))]
+
+    if not condition_list:
+        if export_file is not None:
+            logging.warning('No output file could be written, as no matching '
+                            'preequilibration and simulation condition with '
+                            'steady state data was found.')
```

Comment:
If you return an empty DataFrame it would be more coherent to also write that to file. But preferred to move file I/O out of here (see below).